from PyQt5 import QtCore, QtGui, QtWidgets
import mysql.connector as mc

class Ui_edit(object):
    def setupUi(self, edit):
        edit.setObjectName("edit")
        edit.resize(1129, 665)
        self.centralwidget = QtWidgets.QWidget(edit)
        self.centralwidget.setObjectName("centralwidget")
        self.close = QtWidgets.QPushButton(self.centralwidget)
        self.close.setGeometry(QtCore.QRect(810, 230, 151, 71))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(18)
        self.close.setFont(font)
        self.close.setObjectName("close")

        self.close.clicked.connect(edit.close)

        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(50, 0, 1041, 91))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(48)
        self.label.setFont(font)
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(240, 120, 161, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(240, 200, 151, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_3.setFont(font)
        self.label_3.setAlignment(QtCore.Qt.AlignCenter)
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(240, 360, 71, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_4.setFont(font)
        self.label_4.setAlignment(QtCore.Qt.AlignCenter)
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.centralwidget)
        self.label_5.setGeometry(QtCore.QRect(240, 320, 111, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_5.setFont(font)
        self.label_5.setAlignment(QtCore.Qt.AlignCenter)
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        self.label_6.setGeometry(QtCore.QRect(240, 280, 221, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_6.setFont(font)
        self.label_6.setAlignment(QtCore.Qt.AlignCenter)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        self.label_7.setGeometry(QtCore.QRect(240, 240, 171, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_7.setFont(font)
        self.label_7.setAlignment(QtCore.Qt.AlignCenter)
        self.label_7.setObjectName("label_7")
        self.label_8 = QtWidgets.QLabel(self.centralwidget)
        self.label_8.setGeometry(QtCore.QRect(240, 480, 191, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignCenter)
        self.label_8.setObjectName("label_8")
        self.label_9 = QtWidgets.QLabel(self.centralwidget)
        self.label_9.setGeometry(QtCore.QRect(240, 440, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_9.setFont(font)
        self.label_9.setAlignment(QtCore.Qt.AlignCenter)
        self.label_9.setObjectName("label_9")
        self.label_10 = QtWidgets.QLabel(self.centralwidget)
        self.label_10.setGeometry(QtCore.QRect(240, 400, 171, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_10.setFont(font)
        self.label_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_10.setObjectName("label_10")
        self.label_11 = QtWidgets.QLabel(self.centralwidget)
        self.label_11.setGeometry(QtCore.QRect(240, 520, 81, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_11.setFont(font)
        self.label_11.setAlignment(QtCore.Qt.AlignCenter)
        self.label_11.setObjectName("label_11")
        self.label_12 = QtWidgets.QLabel(self.centralwidget)
        self.label_12.setGeometry(QtCore.QRect(240, 160, 121, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_12.setFont(font)
        self.label_12.setAlignment(QtCore.Qt.AlignCenter)
        self.label_12.setObjectName("label_12")
        self.edit = QtWidgets.QPushButton(self.centralwidget)
        self.edit.setGeometry(QtCore.QRect(810, 150, 151, 71))

        self.edit.clicked.connect(self.edit1)

        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(18)
        self.edit.setFont(font)
        self.edit.setObjectName("edit")
        self.close_2 = QtWidgets.QPushButton(self.centralwidget)
        self.close_2.setGeometry(QtCore.QRect(810, 460, 151, 71))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(18)
        self.close_2.setFont(font)
        self.close_2.setObjectName("close_2")

        self.close_2.clicked.connect(self.delete)

        self.pname = QtWidgets.QLineEdit(self.centralwidget)
        self.pname.setGeometry(QtCore.QRect(400, 120, 261, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.pname.setFont(font)
        self.pname.setObjectName("pname")
        self.notes = QtWidgets.QLineEdit(self.centralwidget)
        self.notes.setGeometry(QtCore.QRect(320, 520, 341, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.notes.setFont(font)
        self.notes.setObjectName("notes")
        self.allergies = QtWidgets.QLineEdit(self.centralwidget)
        self.allergies.setGeometry(QtCore.QRect(350, 320, 311, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.allergies.setFont(font)
        self.allergies.setObjectName("allergies")
        self.add = QtWidgets.QLineEdit(self.centralwidget)
        self.add.setGeometry(QtCore.QRect(430, 480, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.add.setFont(font)
        self.add.setObjectName("add")
        self.disease = QtWidgets.QLineEdit(self.centralwidget)
        self.disease.setGeometry(QtCore.QRect(410, 240, 251, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.disease.setFont(font)
        self.disease.setObjectName("disease")
        self.pid = QtWidgets.QLineEdit(self.centralwidget)
        self.pid.setGeometry(QtCore.QRect(360, 160, 301, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.pid.setFont(font)
        self.pid.setObjectName("pid")
        self.gardname = QtWidgets.QLineEdit(self.centralwidget)
        self.gardname.setGeometry(QtCore.QRect(410, 400, 251, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.gardname.setFont(font)
        self.gardname.setObjectName("gardname")
        self.blood = QtWidgets.QLineEdit(self.centralwidget)
        self.blood.setGeometry(QtCore.QRect(390, 200, 271, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.blood.setFont(font)
        self.blood.setObjectName("blood")
        self.haemoglobin = QtWidgets.QLineEdit(self.centralwidget)
        self.haemoglobin.setGeometry(QtCore.QRect(460, 280, 201, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.haemoglobin.setFont(font)
        self.haemoglobin.setObjectName("haemoglobin")
        self.gardphone = QtWidgets.QLineEdit(self.centralwidget)
        self.gardphone.setGeometry(QtCore.QRect(450, 440, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.gardphone.setFont(font)
        self.gardphone.setObjectName("gardphone")
        self.fees = QtWidgets.QLineEdit(self.centralwidget)
        self.fees.setGeometry(QtCore.QRect(310, 360, 351, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.fees.setFont(font)
        self.fees.setObjectName("fees")
        self.label_13 = QtWidgets.QLabel(self.centralwidget)
        self.label_13.setGeometry(QtCore.QRect(700, 410, 181, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_13.setFont(font)
        self.label_13.setAlignment(QtCore.Qt.AlignCenter)
        self.label_13.setObjectName("label_13")
        self.deletepid = QtWidgets.QLineEdit(self.centralwidget)
        self.deletepid.setGeometry(QtCore.QRect(890, 410, 151, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.deletepid.setFont(font)
        self.deletepid.setObjectName("deletepid")
        edit.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(edit)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1129, 26))
        self.menubar.setObjectName("menubar")
        edit.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(edit)
        self.statusbar.setObjectName("statusbar")
        edit.setStatusBar(self.statusbar)

        self.retranslateUi(edit)
        QtCore.QMetaObject.connectSlotsByName(edit)

    def retranslateUi(self, edit):
        _translate = QtCore.QCoreApplication.translate
        edit.setWindowTitle(_translate("edit", "edit"))
        self.close.setText(_translate("edit", "Close"))
        self.label.setText(_translate("edit", "HOSPITAL MANAGEMENT"))
        self.label_2.setText(_translate("edit", "Patient Name:"))
        self.label_3.setText(_translate("edit", "Blood Group:"))
        self.label_4.setText(_translate("edit", "Fees:"))
        self.label_5.setText(_translate("edit", "Allergies:"))
        self.label_6.setText(_translate("edit", "Haemoglobin Leves:"))
        self.label_7.setText(_translate("edit", "Disease Caused:"))
        self.label_8.setText(_translate("edit", "Patient\'s Address:"))
        self.label_9.setText(_translate("edit", "Gardian Phone No:"))
        self.label_10.setText(_translate("edit", "Gardian Name:"))
        self.label_11.setText(_translate("edit", "Notes:"))
        self.label_12.setText(_translate("edit", "Patient ID:"))
        self.edit.setText(_translate("edit", "Edit"))
        self.close_2.setText(_translate("edit", "Delete"))
        self.label_13.setText(_translate("edit", "Enter Patient ID:"))

    def edit1(self):
        try:

            pname = self.pname.text()
            pid = self.pid.text()
            blood = self.blood.text()
            disease = self.disease.text()
            haemoglobin = self.haemoglobin.text()
            allergies = self.allergies.text()
            fees = self.fees.text()
            gardname = self.gardname.text()
            gardphone = self.gardphone.text()
            patientadd = self.add.text()
            notes = self.notes.text()

            mydb = mc.connect(
                host = 'localhost',
                user='root',
                password='password',
                database='ohms'

            )

            mycursor = mydb.cursor()
            query = "UPDATE overflow SET pname = '"+ pname +"',blood = '"+ blood +"' ,disease = '"+ disease +"' ,haemoglobin = "+ haemoglobin +" ,allergies  = '"+ allergies +"' ,gardname = '"+ gardname +"' ,gardphone = "+ gardphone +" ,patientadd = '"+ patientadd +"' ,notes = '"+ notes +"' WHERE pid = " + pid
            mycursor.execute(query)
            mydb.close()
            print("Successful")

            

        except mc.Error as e:
            print("smth is wrong i can feel it")
            print(query)


    def delete(self):
        try:

            deletepid = self.deletepid.text()
            
            mydb = mc.connect(
                host = 'localhost',
                user='root',
                password='password',
                database='ohms'

            )

            mycursor = mydb.cursor()
            query = "delete from overflow where pid = " + deletepid
            mycursor.execute(query)
            mydb.close()
            print(query)
            print("Successful")
            

            

        except mc.Error as e:
            print("smth is wrong i can feel it")
            print(query)

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    edit = QtWidgets.QMainWindow()
    ui = Ui_edit()
    ui.setupUi(edit)
    edit.show()
    sys.exit(app.exec_())
